internal/server/evaluation/evaluation.go

```diff
@@ -13,21 +16,42 @@ import (
 )
 
 // Variant evaluates a request for a multi-variate flag and entity.
-func (e *EvaluateServer) Variant(ctx context.Context, v *rpcEvaluation.EvaluationRequest) (*rpcEvaluation.VariantEvaluationResponse, error) {
-	e.logger.Debug("evaluate", zap.Stringer("request", v))
+func (s *Server) Variant(ctx context.Context, v *rpcEvaluation.EvaluationRequest) (*rpcEvaluation.VariantEvaluationResponse, error) {
+	var ver = &rpcEvaluation.VariantEvaluationResponse{}
+
+	s.logger.Debug("variant", zap.Stringer("request", v))
 	if v.NamespaceKey == "" {
 		v.NamespaceKey = storage.DefaultNamespace
 	}
 
-	resp, err := e.evaluator.Evaluate(ctx, &flipt.EvaluationRequest{
+	flag, err := s.store.GetFlag(ctx, v.NamespaceKey, v.FlagKey)
+	if err != nil {
+		var errnf errs.ErrNotFound
+
+		if errors.As(err, &errnf) {
+			ver.Reason = rpcEvaluation.EvaluationReason_FLAG_NOT_FOUND_EVALUATION_REASON
+			return ver, err
+		}
+
+		ver.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return ver, err
+	}
+
+	if flag.Type != flipt.FlagType_VARIANT_FLAG_TYPE {
+		ver.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return ver, err
+	}
```

Comment:
If we move the flag existence check up into the server, then I would adjust (both) the evaluator interfaces to take a `*flipt.Flag` on `Evaluate`.
So that the evaluator then doesn't have to evaluate getting it again. Otherwise we're fetching it twice for no reason.